js/lightbeam.js

```diff
@@ -1,9 +1,48 @@
-async function renderGraph() {
-  const canvas = document.getElementById('canvas');
-  const context = canvas.getContext('2d');
-  const websites = await store.getAll();
+function transformData(websites) {
+  // reformat data for D3
+  const nodesData = [];
+  const linksData = [];
+  const thirdParties = [];
+  // make first party nodes
+  for (const firstParty in websites) {
+    const node = {};
+    node.hostname = firstParty;
+    node.favicon = websites[firstParty].faviconUrl || '';
+    node.party = 'first';
+    nodesData.push(node);
+    // make third party nodes and links
+    for (const thirdParty in websites[firstParty].thirdPartyRequests) {
+      // make sure there is only one node per third party
+      if (thirdParties.indexOf(thirdParty) === -1) {
+        // make a third party node
+        thirdParties.push(thirdParty);
+        const node = {};
+        node.hostname = thirdParty;
+        // third parties likely don't have favicons
+        node.favicon = '';
+        node.party = 'third';
+        nodesData.push(node);
+      }
+      const link = {};
+      link.source = firstParty;
+      link.target = thirdParty;
+      linksData.push(link);
+    }
+  }
 
-  viz.draw(context, websites);
+  return [nodesData, linksData];
 }
 
-renderGraph();
+window.onload = async function() {
+  viz.init();
+  const websites = await store.getAll();
+  const data = transformData(websites);
+  // draw the elements onto the page
+  viz.draw(...data);
+};
+
+browser.storage.onChanged.addListener( (changes) => {
```

Comment:
This shouldn't know how storage works. Please leave this out and we can work on this elsewhere.